src/Kiota.Builder/Writers/Java/CodeMethodWriter.cs

```diff
@@ -402,6 +405,25 @@ private void WriteGetterBody(CodeMethod codeElement, LanguageWriter writer, Code
         else
             writer.WriteLine($"return this.{backingStore.Name}.get(\"{codeElement.AccessedProperty?.Name}\");");
     }
+    private void WriteQueryParametersExtractorBody(CodeMethod codeElement, LanguageWriter writer, CodeClass parentClass)
+    {
+        writer.WriteLine("Map<String, Object> allQueryParams = new HashMap();");
+        var allQueryParams = parentClass
+                                .GetPropertiesOfKind(CodePropertyKind.QueryParameter)
+                                .OrderBy(static x => x, CodePropertyTypeForwardComparer)
+                                .ThenBy(static x => x.Name)
+                                .ToArray();
+        foreach (var queryParam in allQueryParams)
+        {
+            var keyValue = queryParam.Name;
+            if (queryParam.IsNameEscaped)
+            {
+                keyValue = queryParam.SerializationName;
+            }
```

Comment:
nit: you could use a ternary expression to avoid an unecessary allocation. but if you prefer to keep it this way for readbility, I won't push back

Reply:
👍 used a ternary operator